### PR TITLE
Fix and improve filters view

### PR DESCRIPTION
**Pull request by jorgenbele**
This patch introduces fixes to the filter view. It
updates it to work correctly with the changes in the API
and also adds functionality for editing existing filters.

All in all this patch seems to work quite well, with
snackbar alerts used to provide the user with feedback.

Doesn't look that good, but does the job.

### CODE REVIEW

**Comment by jorgenbele** [comment]
Closes #158 
Closes #146

File: src/components/filterbuilder/FilterBuilder.tsx [hunk 1]
```diff
@@ -1,66 +1,78 @@
...
 
-import { FilterDefinition, SourceSystem } from "../../api";
-import { toMap, removeUndefined } from "../../utils";
+import { Filter, FilterDefinition, SourceSystem } from "../../api";
+import { removeUndefined } from "../../utils";
 
 import TagSelector, { Tag } from "../../components/tagselector";
 import SourceSelector from "../../components/sourceselector";
 
+import { FiltersContext, FiltersContextType } from "../../components/filters/contexts";
```
Comment (hmpf) [inline src/components/filterbuilder/FilterBuilder.tsx:16]: The "../../components/filters/contexts" file is not included in this commit and does not exist on master.

File: src/views/filters/FiltersView.tsx [hunk 2]
```diff
@@ -8,41 +8,40 @@ import TableCell, { TableCellProps } from "@material-ui/core/TableCell";
...
+import { toMapNum, toMapStr, pkGetter } from "../../utils";
+import TagSelector, { Tag, originalToTag } from "../../components/tagselector";
+import FilteredIncidentTable from "../../components/incidenttable/FilteredIncidentTable";
 
-import { pkGetter, toMap } from "../../utils";
-
-import IncidentsPreview from "../../components/incidentspreview/IncidentsPreview";
+import "../../components/incidenttable/incidenttable.css";
 
+import { FiltersContext, FiltersContextType, DEFAULT_FILTERS_CONTEXT_VALUE } from "../../components/filters/contexts";
```
Comment (hmpf) [inline src/views/filters/FiltersView.tsx:27]: The "../../components/filters/contexts" file is not included in this commit and does not exist on master.

**Review by hmpf — CHANGES_REQUESTED** [review]
Is a file missing? `src/components/filters/contexts.tsx` or something?

**Comment by hmpf** [comment]
I'll hand-test..

**Review by hmpf — COMMENTED** [review]
I can't make the filter-bits of this this talk to the backend properly.

**Review by hmpf — APPROVED** [review]
Tested and verified against backend v1.0.0.

**Pull request merged**